Content.Server/Power/Events/BatteryChangedEvent.cs

```diff
@@ -0,0 +1,7 @@
+namespace Content.Server.Power.Events;
+
+/// <summary>
+/// Raised on an entity when a related entity that stores a powercell has its powercell changed.
+/// </summary>
+[ByRefEvent]
+public record struct BatteryChangedEvent(EntityUid Battery, EntityUid BatteryHolder);
```

Comment:
unless you're going to somehow integrate this with the normal powercell system, this should just be renamed and kept specifically for ninja.